chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -260,7 +259,11 @@ public synchronized void set(int type, int x, int y, int z) {
     } catch(PackedOctree.OctreeTooBigException e) {
       // Octree is too big, switch implementation and retry
       Log.warn("Octree is too big, falling back to old (slower and bigger) implementation.");
-      switchToNodeBased();
+      try {
+        switchImplementation("NODE");
+      } catch(IOException ioException) {
+        ioException.printStackTrace();
```

Comment:
You should rethrow this as a runtime exception (`throw new RuntimeException(e, "Couldn't switch the octree implementation to NODE")`).